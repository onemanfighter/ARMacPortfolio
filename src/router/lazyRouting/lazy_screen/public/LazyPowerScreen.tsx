import { lazy } from 'react';
import { LazyProvider } from '../../component';

const Power = lazy(() => import('../../../../screens/public/Power/Power'));

const LazyPowerComponent = () => {
  return (
    <LazyProvider>
      <Power />
    </LazyProvider>
  );
};

export { LazyPowerComponent };
