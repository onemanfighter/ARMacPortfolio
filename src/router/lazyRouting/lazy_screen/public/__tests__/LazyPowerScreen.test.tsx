import { render, screen, waitFor } from '@testing-library/react';
import { LazyPowerComponent } from '../LazyPowerScreen';

describe('LazyPowerComponent', () => {
  it('should render correctly to match snapshot', async () => {
    const { container } = render(<LazyPowerComponent />);

    await waitFor(() => expect(screen.getByText('Power')).toBeDefined());
    expect(container).toMatchSnapshot();
  });
});
