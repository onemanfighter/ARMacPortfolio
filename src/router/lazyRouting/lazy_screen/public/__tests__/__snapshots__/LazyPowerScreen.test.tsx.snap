// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyPowerComponent should render correctly to match snapshot 1`] = `
<div>
  <div>
    <h1>
      Power
    </h1>
  </div>
</div>
`;
